dotCMS/src/main/java/com/dotmarketing/db/commands/UpsertCommand.java

```diff
@@ -0,0 +1,306 @@
+
+import com.dotcms.repackage.org.apache.commons.lang.StringUtils;
+import com.dotcms.system.SimpleMapAppContext;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.common.util.SQLUtil;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.db.DbType;
+import com.dotmarketing.exception.DotDataException;
+import com.liferay.util.StringUtil;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Generic UpsertCommand that can be used to Generate and execute Native SQL Upsert Queries
+ * @author Andre Curione
+ */
+public abstract class UpsertCommand implements DatabaseCommand {
+
+    /**
+     * Map that contains Upsert Command implementations
+     */
+    private static Map<DbType, UpsertCommand> commandMap = new ConcurrentHashMap<>();
+
+    static {
+        commandMap.put(DbType.H2, new H2UpsertCommand());
+        commandMap.put(DbType.POSTGRESQL, new PostgreUpsertCommand());
+        commandMap.put(DbType.MYSQL, new MySQLUpsertCommand());
+        commandMap.put(DbType.MSSQL, new MSSQLUpsertCommand());
+        commandMap.put(DbType.ORACLE, new OracleUpsertCommand());
+    }
+
+    public static UpsertCommand getInstance() {
+        return commandMap.get(DbType.getDbType(DbConnectionFactory.getDBType()));
+    }
+
+    /**
+     * Method to execute The Upsert Query... This method works for most DBTypes,
+     * and its overriden for DBTypes with particularities
+     * @param dotConnect Connection
+     * @param query SQL String to be executed
+     * @param parameters or values
+     * @throws DotDataException
+     */
+    public void execute(DotConnect dotConnect, String query, Object... parameters)
+            throws DotDataException {
+        DotConnect dc = (dotConnect != null) ? dotConnect : new DotConnect();
+        ArrayList<Object> params =  new ArrayList<>();
+        Collections.addAll(params, parameters); //Update parameters
+        Collections.addAll(params, parameters); //Insert parameters
+        dc.executeUpdate(query, params.toArray());
+    }
+
+    // Misc Methods:
+    /**
+     * Generates the Update statement query part, with the format:
+     * column = ?, column = ?, etc.
+     */
+    protected String getUpdateColumnValuePairs(SimpleMapAppContext replacements) {
+        StringBuilder builder = new StringBuilder();
+        if (replacements.getAttribute(QueryReplacements.ID_COLUMN) != null) {
+            builder.append(replacements.getAttribute(QueryReplacements.ID_COLUMN).toString());
+            builder.append("=");
+            builder.append(replacements.getAttribute(QueryReplacements.ID_VALUE).toString());
+            builder.append(",");
+        }
+        if (replacements.getAttribute(QueryReplacements.CONDITIONAL_COLUMN) != null) {
+            if (!(this instanceof OracleUpsertCommand)) { //Oracle does not allow to Update the Conditional Column
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)